tools/bots/mission-control-bot/actions.ts

```diff
@@ -0,0 +1,356 @@
+import { MessageEmbed, Message } from "discord.js";
+import Web3 from "web3";
+import { TransactionReceipt } from "web3-core";
+import PromiEvent from "web3-core-promievent";
+
+import { Receivers, WorkerAccount, FundsRequest } from "./types";
+import { TOKEN_DECIMAL, EMBED_COLOR_CORRECT, EMBED_COLOR_ERROR, params } from "./constants";
+import { sendSlackNotification, nextAvailableToken, sleep } from "./utils";
+
+/**
+ * Action for the bot for the pattern "!faucet send <h160_addr>", that
+ * sends funds to the indicated account.
+ * @param web3Api Instance of the web3 API connected to the chain endpoint
+ * @param msg Received discord message object
+ * @param discordUserId Author ID of the message
+ * @param messageContent Content of the message
+ * @param discordUserReceivers Map with the timestamp of the last received request of a user
+ * @param pendingQueue Queue of tasks
+ */
+export async function botActionFaucetSend(
+  web3Api: Web3,
+  msg: Message,
+  discordUserId: string,
+  messageContent: string,
+  discordUserReceivers: Receivers,
+  addressReceivers: Receivers,
+  pendingQueue: FundsRequest[]
+) {
+  // get address from message
+  let address = messageContent.slice("!faucet send".length).trim();
+  if (!address.startsWith("0x")) address = `0x${address}`;
+
+  // check address and send alert msg and return if bad formatted
+  if (!web3Api.utils.isAddress(address)) {
+    const errorEmbed = new MessageEmbed()
+      .setColor(EMBED_COLOR_ERROR)
+      .setTitle("Invalid address")
+      .setFooter("Addresses must follow the H160 address format");
+
+    // send message to channel
+    msg.channel.send(errorEmbed);
+    // exit
+    return;
+  }
+
+  // set default value of lastReceived at 0
+  if (!discordUserReceivers[discordUserId]) discordUserReceivers[discordUserId] = 0;
+  if (!addressReceivers[address]) addressReceivers[address] = 0;
+
+  let reason = null;
+  const canReceiveTokensAgain = await (async function (): Promise<boolean> {
```

Comment:
Why use this await `(async function()....)()` syntax ? It doesn't look async

If you want to keep a single assignment logic, you can do
```
const lastPermittedClaimDate= Date.now() - params.FAUCET_SEND_INTERVAL * 3600 * 1000 ;
    
const [reason, canReceiveTokensAgain] = params.NOT_LIMITED_USERS.includes(discordUserId) ?
  [null, true] : ? discordUserReceivers[discordUserId] > lastPermittedClaimDate ?
  ["user", false] ?  addressReceivers[address] > lastPermittedClaimDate ?
  ["address", false] : [null, true]